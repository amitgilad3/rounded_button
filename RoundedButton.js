import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, Text, View } from 'react-native'
import styles from './styles'

export default class RoundedButton extends Component {
  static propTypes = {
    onPress: PropTypes.func,
    label: PropTypes.string,
    style: PropTypes.oneOfType([PropTypes.object, PropTypes.number])
  }

  render () {
    const { onPress, label, style, testID } = this.props

    return (
      <View testID={testID} style={[styles.buttonContainer, style]}>
        <TouchableOpacity style={styles.button} onPress={onPress}>
          <Text style={styles.buttonText}>{label}</Text>
        </TouchableOpacity>
      </View>
    )
  }
}
