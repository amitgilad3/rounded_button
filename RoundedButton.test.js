import 'react-native'
import React from 'react'
import RoundedButton from './RoundedButton'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('<RoundedButton />', () => {
  it('RoundedButton component renders correctly', () => {
    const tree = renderer
      .create(<RoundedButton onPress={() => {}} text='howdy' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('RoundedButton component with children renders correctly', () => {
    const tree = renderer
      .create(<RoundedButton onPress={() => {}}>Howdy</RoundedButton>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles onPress callback', () => {
    let i = 0 // i guess i could have used sinon here too... less is more i guess
    const onPress = () => i++
    const wrapperPress = shallow(<RoundedButton onPress={onPress} text='hi' />)
    const touchable = wrapperPress.find('TouchableOpacity')

    expect(touchable.prop('onPress')).toBe(onPress) // uses the right handler
    expect(i).toBe(0)
    touchable.simulate('press')
    expect(i).toBe(1)
  })
})
