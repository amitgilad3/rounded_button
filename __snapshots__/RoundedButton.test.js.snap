// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundedButton /> RoundedButton component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 24,
        "elevation": 24,
        "height": 48,
        "marginHorizontal": 25,
        "marginVertical": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
  testID={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#DE2C2E",
          "fontSize": 14,
          "fontWeight": "900",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`<RoundedButton /> RoundedButton component with children renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 24,
        "elevation": 24,
        "height": 48,
        "marginHorizontal": 25,
        "marginVertical": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 16,
      },
      undefined,
    ]
  }
  testID={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#DE2C2E",
          "fontSize": 14,
          "fontWeight": "900",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
